sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -109,6 +116,7 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
         case LongType => tryParseLong(field)
         case _: DecimalType => tryParseDecimal(field)
         case DoubleType => tryParseDouble(field)
+        case DateType => tryParseDateFormat(field)
```

Comment:
Just curious why you try to infer dates before timestamps?

Reply:
This is as per the suggestion in one of the review comments to use

https://github.com/apache/spark/pull/32558#discussion_r633020781